libraries-3/src/test/java/com/baeldung/jnats/NatsClientLiveTest.java

```diff
@@ -1,115 +1,128 @@
 package com.baeldung.jnats;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
 
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
 
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
 
 import io.nats.client.Message;
-import io.nats.client.SyncSubscription;
+import io.nats.client.Subscription;
 
+/**
+ * All the tests in this class require that a NATS server be running on localhost at the default port.
+ * See {@link <a href="https://docs.nats.io/nats-server/installation">Installing a NATS Server</a>}.
+ */
 public class NatsClientLiveTest {
 
-    @Test
-    public void givenMessageExchange_MessagesReceived() throws Exception {
-
-        NatsClient client = connectClient();
-
-        SyncSubscription fooSubscription = client.subscribeSync("foo.bar");
-        SyncSubscription barSubscription = client.subscribeSync("bar.foo");
-        client.publishMessage("foo.bar", "bar.foo", "hello there");
+    public static final int TIMEOUT_MILLIS = 200;
 
-        Message message = fooSubscription.nextMessage(200);
-        assertNotNull("No message!", message);
-        assertEquals("hello there", new String(message.getData()));
-
-        client.publishMessage(message.getReplyTo(), message.getSubject(), "hello back");
-
-        message = barSubscription.nextMessage(200);
-        assertNotNull("No message!", message);
-        assertEquals("hello back", new String(message.getData()));
+    private NatsClient connectClient() throws IOException, InterruptedException {
+        return new NatsClient(NatsClient.createConnection("nats://localhost:4222"));
     }
 
-
-    private NatsClient connectClient() {
-        return new NatsClient();
+    @Test
+    public void givenMessageExchangeViaPublishSubscribe_MessagesReceived() throws Exception {
```

Comment:
Test names still not matching BDD syntax

Reply:
Will fix